zebra-chain/src/sapling/spend.rs

```diff
@@ -161,7 +161,7 @@ impl Spend<SharedAnchor> {
 impl ZcashSerialize for Spend<PerSpendAnchor> {
     fn zcash_serialize<W: io::Write>(&self, mut writer: W) -> Result<(), io::Error> {
         self.cv.zcash_serialize(&mut writer)?;
-        writer.write_all(&self.per_spend_anchor.0[..])?;
+        writer.write_all(&self.per_spend_anchor.0.to_bytes())?;
```

Comment:
Optional:  I think this might work because you implemented `ZcashSerialize for Root`:

```suggestion
        self.per_spend_anchor.zcash_serialize(&mut writer)?;
```

Reply:
https://github.com/ZcashFoundation/zebra/pull/3544/commits/6a8b4b8a18f4206da6e712cf6fc31d8ee5a55c78